src/test/java/seedu/jarvis/model/quest/QuestListTest.java

```diff
@@ -1,18 +1,75 @@
 package seedu.jarvis.model.quest;
 
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static seedu.jarvis.testutil.Assert.assertThrows;
+import static seedu.jarvis.testutil.TypicalQuests.COLORFUL_CARPETS;
+import static seedu.jarvis.testutil.TypicalQuests.RUNIC_CARPETS;
+
+import java.util.ArrayList;
+import java.util.List;
+
 import org.junit.jupiter.api.Test;
 
+import seedu.jarvis.testutil.QuestBuilder;
+
 class QuestListTest {
 
+
+    private final QuestList questList = new QuestList();
+    private final List<Quest> questListTest = new ArrayList<>();
+
+    @Test
+    void add_nullQuest_throwsNullPointerException() {
```

Comment:
This type of checking is indeed really important, thank you for writing these tests cases!